src/Mod/TechDraw/Gui/ViewProviderDrawingView.cpp

```diff
@@ -234,37 +240,50 @@ void ViewProviderDrawingView::onGuiRepaint(const TechDraw::DrawView* dv)
 
     std::vector<TechDraw::DrawPage*> pages = getViewObject()->findAllParentPages();
     if (pages.size() > 1) {
-        for (auto& p : pages) {
-            std::vector<App::DocumentObject*> views = p->Views.getValues();
-            for (auto& v: views) {
-                if (v == getViewObject()) {
-                    //view v belongs to this page p
-                    ViewProviderPage* vpPage = getViewProviderPage();
-                    if (vpPage) {
-                        if (vpPage->getQGSPage()) {
-                            QGIView* qView = dynamic_cast<QGIView *>(vpPage->getQGSPage()->findQViewForDocObj(v));
-                            if (qView != nullptr) {
-                                qView->updateView(true);
-                            }
-                        }
-                    }
+        multiParentPaint(pages);
+    } else if (dv == getViewObject()) {
+        singleParentPaint(dv);
+    }
+}
+
+void ViewProviderDrawingView::multiParentPaint(std::vector<TechDraw::DrawPage*>& pages)
+{
+    for (auto& p : pages) {
+        std::vector<App::DocumentObject*> views = p->Views.getValues();
+        for (auto& v: views) {
+            if (v != getViewObject()) {  //should this be dv from onGuiRepaint?
+                continue;
+            }
+            //view v belongs to this page p
+            ViewProviderPage* vpPage = getViewProviderPage();
+            if (!vpPage) {
+                continue;
+            }
+            if (vpPage->getQGSPage()) {
+                QGIView* qView = dynamic_cast<QGIView *>(vpPage->getQGSPage()->findQViewForDocObj(v));
+                if (qView) {
+                    qView->updateView(true);
                 }
             }
         }
-    } else if (dv == getViewObject()) {
-        //original logic for 1 view on 1 page
-        if (!dv->isRemoving() &&
-            !dv->isRestoring()) {
-            QGIView* qgiv = getQView();
-            if (qgiv) {
-                qgiv->updateView(true);
-            } else {       //we are not part of the Gui page yet. ask page to add us.
-                ViewProviderPage* vpPage = getViewProviderPage();
-                if (vpPage) {
-                    if (vpPage->getQGSPage()) {
-                        vpPage->getQGSPage()->addView(dv);
-                    }
-                }
+    }
+}
+
+void ViewProviderDrawingView::singleParentPaint(const TechDraw::DrawView* dv)

```

Comment:
Split up in functions make it possible to both read the code the function was called by and also decode what the purpose of the algorithm is, since it is described in the functions name
😍